### DISCUSSION TITLE
Background apps subtitle

### DISCUSSION THREAD

**Discussion by diegopvlk** [andyholmes/valent#545] [Feature Request]
**First Request:**
_Edit: nevermind, it was actually my phone, there's a second setting about the battery_ 

**Second Request:**
A subtitle for the background apps list with the device name and battery:
"deviceName ⦁ 76%"

**Comment by andyholmes** [comment]
I don't think think the background apps subtitle is really possible to make dynamic, unfortunately. It would also probably require the plugin to poke around outside of its regular scope, which is a bit weird.

The GNOME Shell extension exposes such information in the UI, however. I'm sure it wouldn't be too hard to do the same for other DE's if necessary.

**Reply by not-a-dev-stein** [reply]
Would it be possible for it to show whenever the device is connected to it or not? It could be something useful for the subtitle while (potentially) not needing a ton of moving parts to maintain.

**Reply by andyholmes** [reply]
It's not about moving parts, really. The subtitle is the `reason` string you provide for [`RequestBackground()`](https://docs.flatpak.org/en/latest/portal-api-reference.html#gdbus-method-org-freedesktop-portal-Background.RequestBackground).

Using it that way would mean re-requesting the permission repeatedly, and the "reason" presented to the user would be "Connected"/"Disconnected".

**Reply by not-a-dev-stein** [reply]
Oh, shame it isn't possible, it'd be great to at least find some string like "Syncing with device" (or $DEVICE with the device name), but yeah, it doesn't seem like a feature important enough to lose sleep for.

**Reply by andyholmes** [reply]
Well, there can be more than one device, too :) But I'm not opposed to a static subtitle, if there's one that makes sense.

**Reply by not-a-dev-stein** [reply]
Oh, that could be a nice brainstorming thing. What about something like "Syncing your device" instead, focusing on the fact that the machine that is running Valent is using it to sync information?

**Reply by andyholmes** [reply]
Maybe "Syncing Devices"?

**Reply by not-a-dev-stein** [reply]
Yeah, that's a lot better, works for multiple and single device connections, since it can mean "connecting your current device and the one you paired with it"

**Reply by andyholmes** [reply]
Ah, apparently this can be somewhat dynamic. I think I'm going to stick with "Syncing Devices" for now, though.

Is there a reason to want the status to be dynamic, rather than using the GNOME Shell extension or other plugin system?

**Reply by not-a-dev-stein** [reply]
> Is there a reason to want the status to be dynamic, rather than using the GNOME Shell extension or other plugin system?

There can be plenty of potential use cases for it, specially to know at a glance:

- if the device you are paired with is connected or not ("Connected" or "Disconnected")
- which devices are connected ("Connected to $DEVICE")
- how many devices are connected ("Connected to X devices")
- the battery status of the connected device, if there's only one.

The last one is specially useful to know at a glance, since the shell extension only gives this information (AFAIK, I haven't used it in quite a while) if you use it on the system tray instead of the quick settings. So you could (in the dream scenario) have something similar to @diegopvlk original request:

If no devices are currently connected the message would be "Disconnected" or "Looking for device";
If connected to one device show the name and battery ("Syncing to $deviceName ⦁ 76%"), this probably would break the character limit though (if it has one in the first place, not sure) depending on the name of the device.
If connected to multiple devices, show how many devices you are connected to ("Syncing to X devices").

**Reply by andyholmes** [reply]
Hmm, yeah that's quite a bit more information than I'm comfortable reaching for from the XDP plugin.

From the Quick Settings you can see the name of a single connected device or how many devices are connected, and open it for battery information. Each device menu item can be used to open that device's page as well.

The background apps menu won't show you anything until you open it, and obviously can't have icons, sub-items or much of anything than a single line of text. I'm not sure it's really suited for this sort of thing.

**Reply by not-a-dev-stein** [reply]
I've been using it for a while now and the background subtitle works great, with just one small issue. When you first turn on the PC and Valent starts in the background, it doesn't show it at all. Only after opening the app once and closing it again will it show the subtitle.

That said, I have to ask, do you consider the app stable enough at this point to recommend over the extension or GSConnect? I've been wanting to make a post for Fedora's social media recommending it, but I'm not sure what the situation with the app is, beyond that it's currently on Nightly builds instead of available on Flathub.

**Reply by andyholmes** [reply]
> Only after opening the app once and closing it again will it show the subtitle.

Yes, I believe it's still required for an application to have a visible window to request background permissions. Although I guess this is probably only necessary the first time; you could open a bug if you like.

> That said, I have to ask, do you consider the app stable enough at this point to recommend over the extension or GSConnect?

If you mean `gnome-shell-extension-valent`, that's an addition for Valent, not an alternative. As for Valent vs. GSConnect, the status I posted [here](https://github.com/andyholmes/valent/discussions/163) is still pretty accurate.

**Reply by not-a-dev-stein** [reply]
> If you mean gnome-shell-extension-valent, that's an addition for Valent, not an alternative.

Oh sorry, I mistyped and it sounded completely wrong, I meant as a comparison with GSConnect.

> As for Valent vs. GSConnect, the status I posted https://github.com/andyholmes/valent/discussions/163 is still pretty accurate.

Makes sense. I'm personally biased in favour of recommending apps instead of extensions, so if you're ok with it I can create a post about it for our social media. I really like the project and would love to get more eyes its way.

**Reply by andyholmes** [reply]
Sure, I won't have much time to dedicate to it for awhile though, so as long as that's okay.
